http_server.go

```diff
@@ -189,6 +189,15 @@ func (hs *HTTPServer) configureHTTP2() error {
 	return hs.config.ConfigureTLSConfig(hs.srv.TLSConfig)
 }
 
+func (hs *HTTPServer) BindPort() error {
+	if hs.Listener != nil {
+		return fmt.Errorf("HTTP server already bound to port")
+	}
+
+	_, err := hs.listener()
+	return err
+}
```

Comment:
I believe, it's not going to work as we neither release the port, nor set `hs.Listener`.

Reply:
I see two approaches there:
1. ListenerDryRun -  try to bind and release the port.
2. Manual listener start - like you did but set hs.Listener.

Reply:
Ugh, the commit I meant to push included 9735029

Reply:
Binding and releasing the port won't work as it may use `0` to bind a random port, and then we introduce a race condition when reusing.

Reply:
I'm not sure I understand the problem. Why do we need this?
If the server fails to bind, we fail, no need to check for it, in advance